src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs

```diff
@@ -1,119 +1,101 @@
 			{
-				headerUntapZone.PointerPressed -= OnTap;
-			}
+				_headerUntapZone = _timePickerPresenter?.FindName("HeaderUntapableZone") as FrameworkElement;
 
-			var timeSelector = Content as TimePickerSelector;
+				AttachAcceptCommand(_timePickerPresenter);
+				AttachDismissCommand(_timePickerPresenter);
 
-			if (timeSelector != null)
+				if (_timePickerPresenter != null)
+				{
+					_timePickerPresenter.Loaded -= onLoad;
+				}
+			}
+
+			if (_timePickerPresenter != null)
 			{
-				timeSelector.Cancel();
+				_timePickerPresenter.Loaded += onLoad;
 			}
 
-			base.Close();
+			return _timePickerPresenter;
 		}
 
+		private void OnTap(object sender, Input.PointerRoutedEventArgs e) => e.Handled = true;
+
 		protected internal override void Open()
 		{
-			base.Open();
+			_timeSelector?.Initialize();
 
-			var timePickerPresenter = _presenter as TimePickerFlyoutPresenter;
-			var headerUntapZone = timePickerPresenter?.FindName("HeaderUntapableZone") as FrameworkElement;
-
-			if (headerUntapZone != null)
+			//Gobbling pressed tap on the flyout header background so that it doesn't close the flyout popup. 
+			if (_headerUntapZone != null)
 			{
-				//Gobbling pressed tap on the flyout header background so that it doesn't close the flyout popup. 
-				headerUntapZone.PointerPressed += OnTap;
+				_headerUntapZone.PointerPressed += OnTap;
 			}
-		}
-
-		private void InitializeContent()
-		{
-			Content = new TimePickerSelector()
-			{
-				BorderThickness = Thickness.Empty,
-				HorizontalAlignment = HorizontalAlignment.Stretch,
-				HorizontalContentAlignment = HorizontalAlignment.Stretch
-			};
 
-			BindToContent(nameof(Time));
-			BindToContent(nameof(ClockIdentifier));
+			base.Open();
 		}
 
-		protected override Control CreatePresenter()
+		protected internal override void Close()
 		{
-			var timePickerPresenter = new TimePickerFlyoutPresenter() { Content = Content };
-
-			void onLoad(object sender, RoutedEventArgs e)
-			{
-				AttachAcceptCommand(timePickerPresenter);
-				AttachDismissCommand(timePickerPresenter);
-				timePickerPresenter.Loaded -= onLoad;
-			}
-
-			if (timePickerPresenter != null)
+			if (_headerUntapZone != null)
 			{
-				timePickerPresenter.Loaded += onLoad;
+				_headerUntapZone.PointerPressed -= OnTap;
 			}
 
-			return timePickerPresenter;
-		}
+			_timeSelector?.Cancel();
 
-		private void BindToContent(string propertyName)
-		{
-			this.Binding(propertyName, propertyName, Content, BindingMode.TwoWay);
+			base.Close();
 		}
 
-		private void AttachAcceptCommand(IFrameworkElement rootControl)
+		private void AttachAcceptCommand(IFrameworkElement control)
 		{
-			var acceptButton = rootControl.FindName("AcceptButton") as Button;
-
-			if (acceptButton != null && acceptButton.Command == null)
+			if (control?.FindName("AcceptButton") is Button b && b.Command == null)
 			{
-				acceptButton.Command = new DelegateCommand(Accept);
+				b.Command = new DelegateCommand(Accept);
```

Comment:
This will also leak, in a subtle way. The delegate will be kept alive by the `b` reference.

Reply:
control?.FindName("DismissButton") is Button b 

is same as

var b = control?.FindName("DismissButton") as Button

can you explain where will be the memory leak?

Reply:
The `Accept` delegate instance contains a reference to `this`, which will be stored in `b`, making a circular reference. You can create a weakference to `this` then do `() => (wr.Target as TimePickerFlyout)?.Accept()`